### ISSUE TITLE
Repeater for SCHEDULED keyword - auto advancing into the future

### ISSUE THREAD

**Issue by fosshelper** [orgzly/orgzly-android#397]
I have searched for existing issues that may be the same as or related to mine. 👼

I'm submitting a **feature request**.
This might also be a question. So for now, if I set a TODO with a repeated timer, then clicking DONE advances the time of the TODO. Suppose I have an appointment added with a SCHEDULED field, with a repeated timer (for example, weekly lessons, etc), is there a way Orgzly can (or already does?) automatically advance the times of those sessions to be in the future? I am asking with special regard to the `++` operator in specifying timestamps.

I know I can specify them as TODO and manually mark them as DONE everytime, but they are not really todos, so... I also sync the org files with my desktop emacs, so if someone knows of related emacs packages/org functions then let me know (in which case, I can put the duty of advancing times into the future on my computer).

**Comment by nevenz** [comment]
What do you need this for? For reminders, agenda view, or... ?

I'm not sure it's a good idea to to these kind of modifications automatically. I'd much rather support whatever would give a similar effect (new query option, new reminder flag), then modify notes.

**Comment by fosshelper** [comment]
I wanted it for agenda view, and in particular, for the searches I have set up for the widget on the homescreen. It's a bit of a stretch, as sort of the same things can be marked with a `TODO`. But its just that certain events will just happen regardless if you mark them as `DONE`. For those types of events, this makes some sense.

**Comment by nevenz** [comment]
Is it to avoid displaying the past entries and just display those in the future? As that could be handled by a new query option. For example #330 is about a similar type of problem - in that case it's too many feature events displayed.

Marking notes as done automatically would raise bunch of questions - when and how often to do it, should auto-sync be triggered (as notebook is modified), do we log time shift like it's done now with "Log to drawer" option on, or do we need a new option to control this separately, etc.

**Comment by fosshelper** [comment]
Well, its not _just_ for hiding past entries.
Let me elaborate by means of an example. Suppose you have a group meeting every Wed (say a biology class, or a weekly TV programme, etc) and it will happen regardless of if _you_ do it (regardless of if you "mark it as done" since it is not a TODO). Then one could set up a weekly repeater in orgmode. As it is not a todo, it doesn't get a `TODO` tag, but does get the `SCHEDULED` property. But it has to be updated each time an event is over, right? For the next week. And so on.

This can certainly be accomplished by setting it as a TODO and marking it DONE manually. However, as explained above, it is _not quite_ a TODO. Regarding other things, the app can do it hourly (or daily), auto-sync to me should be triggered (but then again, I am using it with Dropbox, so no autosync there). Lastly I don't log to drawer anyway, and especially wouldn't care for such `SCHEDULED` events, as they happen regardless of anything else.

**Comment by fosshelper** [comment]
Yes, I wanted it to be displayed every week. You are right in that it is more suited to have a plain timestamp, but since Orgzly doesn't support plain timestamps yet, I set up a SCHEDULED tag.

> But I think that in general it's OK for notes with repeater to have "starting date" in the past 
> and still be relevant.

I agree with this. I see what you are saying. Maybe I can use agenda with this then. Without the agenda search query, the next repeat won't be visible in that way. In any case, that is not implemented yet in Orgzly perhaps.

**Comment by nevenz** [comment]
> I agree with this. I see what you are saying. Maybe I can use agenda with this then. Without the agenda search query, the next repeat won't be visible in that way. In any case, that is not implemented yet in Orgzly perhaps.

Oh I though you said you're using the agenda. Yes, without `ad.DAYS` there is no generating of multiple time instances going on.

It would be possible to add a query option to do that until the next displayed instance, but I'm not sure if that would make sense. Depends what your non-agenda query looks like I guess.

**Comment by SanskritFritz** [comment]
I have tried to set up a daily alarm in Orgzly with ```SCHEDULED: ++1d```. When the scheduled time comes, a notification appears, and I have to advance the scheduled date manually. Sometimes I forget that, I just close the notification, hence the scheduled time is not advanced automatically. Is there a way to advance it automatically whenever it reaches the scheduled time?
